### PR TITLE
PR: Handle empty options in pycodestyle and pydocstyle preferences

### LINKED ISSUE #9209: Setting ignore rules for Pycodestyle is not working
While working on PR #9186 I noticed that when you set something to ignore in the "Code style" tab of "Completion and linting", that's not getting send to server.

@dalthviz, I think the problem is here

https://github.com/spyder-ide/spyder/blob/5a39ace4a24d95b0b4ecaf0fada0317988d37dc3/spyder/plugins/editor/lsp/manager.py#L245

because:

1. In case 'pycodestyle/select' is an empty string, then this code generates `['']` to pass to the server and it should be `[]` (i.e. an empty list, not a list with an empty string).
2. I think Pycodestyle doesn't accept that both `select` and `ignore` are passed at the same time (but please verify that).

Also, please remove the `xfail` of this test after fixing this issue:

https://github.com/spyder-ide/spyder/blob/d3ca22ab0fbe99ed4d64efcb3ee78b13e893452e/spyder/plugins/editor/widgets/tests/test_warnings.py#L27-L32

### PR DESCRIPTION
**Pull request by dalthviz**
## Description of Changes

* [ ] Wrote at least one-line docstrings (for any new functions)
* [ ] Added unit test(s) covering the changes (if testable)
* [ ] Included a screenshot or animation (if affecting the UI, see [Licecap](https://www.cockos.com/licecap/))

### Issue(s) Resolved

Fixes #9209 

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

I certify the above statement is true and correct: dalthviz

### CODE REVIEW

**Comment by dalthviz** [comment]
@ccordoba12 testing the behavior of the two options (`select` and `ignore`), what I see is that when you use the `select` option, only the warnings selected are shown (the other ones are ignore). Also, the `select` option takes precedence to the `ignore` option, so for example if you put for the `select` and `ignore` options the `E202` code, only the warnings associted with an `E202` warning will be shown

File: spyder/plugins/editor/lsp/manager.py [hunk 1]
```diff
@@ -248,10 +248,11 @@ def generate_python_config(self):
...
             'enabled': self.get_option('pycodestyle'),
-            'exclude': [exclude.strip() for exclude in cs_exclude],
-            'filename': [filename.strip() for filename in cs_filename],
-            'select': [select.strip() for select in cs_select],
-            'ignore': [ignore.strip() for ignore in cs_ignore],
+            'exclude': [exclude.strip() for exclude in cs_exclude if exclude],
+            'filename': [filename.strip()
+                         for filename in cs_filename if filename],
+            'select': [select.strip() for select in cs_select if select],
+            'ignore': [ignore.strip() for ignore in cs_ignore if ignore],
```
Comment (ccordoba12) [inline spyder/plugins/editor/lsp/manager.py]: Pretty nice usage of list comprehension! :+1:

**Comment by ccordoba12** [comment]
Merging because `pycodestyle` accepts both `select` and `ignore`, as @dalthviz pointed out.

The question about mentioning this in our docs is a separate issue.

**Comment by CAM-Gerlach** [comment]
Thanks, @ccordoba12 . I created spyder-ide/spyder-docs#96 to cover this.

**Pull request merged**